### DISCUSSION TITLE
Project Status/Is it worth my time making a PR with a feature

### DISCUSSION THREAD

**Discussion by Pieloaf** [dstotijn/hetty#125] [General]
Im just wondering what the status of this project is as I see the last commit and release was 18 months ago, but I see you were recently active in a discussion just last week, so Im unsure where it's at. 

I'd like to add functionality for automatically modifying intercepted requests/responses which match certain parameters. Similar to what was suggested in https://github.com/dstotijn/hetty/discussions/40. I already have my own node js script setup for this locally and thought about building a UI around it but seeing as this project already exists, I thought building it into hetty could be nicer solution but if the project is no longer maintained and the PR could never be merged or if its not a feature you're interested, I probably wouldn't go to the effort of adding it here and take it elsewhere. 

If you would consider this feature for hetty I would make a design proposal to make sure youre ok with the architecture/design of the implementation before working on it.

**Comment by dstotijn** [comment]
Hi @Pieloaf, at the moment there is no active/ongoing development, due to time constraints in my personal and working life. However, I'm still ambitious to eventually pick things up, but cannot give any concrete indication when. Realistically speaking, you might want to explore [Zap's automation framework](https://www.zaproxy.org/docs/desktop/addons/automation-framework/) instead for your personal use case, either to use as-is or to contribute towards.

As for Hetty it's worth noting is that since [v0.7.0](https://github.com/dstotijn/hetty/releases/tag/v0.7.0) there's support for (non-automated) intercept and modify behavior. Some of the foundational work(the middleware architecture to hook into the lifecycle of an HTTP request) has been done as part of that release already. But the design and implementation of a plugin architecture is still "to do": If you have a moment, any details on your use case for automation and ideas/input on the design is greatly appreciated. There's [this issue](https://github.com/dstotijn/hetty/issues/26) on the roadmap to track this.
